adr/0016-compliance-in-pepr.md

```diff
@@ -0,0 +1,51 @@
+# 14. Compliance in Pepr 
+
+Date: 2025-02-13
+
+
+## Status
+
+Pending
+
+
+## Context
+
+Pepr needs to generate reports for compliance purposes, demonstrating adherence to required security and compliance standards through specific bindings. To achieve this, it’s essential to map compliance controls to Pepr’s bindings. During our discovery process, we determined that Pepr’s role in compliance should focus on reporting rather than enforcement. This is because compliance controls are already enforced through the bindings themselves, and Pepr’s primary function is to report on the compliance status of resources.
+
+## Decision
+
+Extend the fluent API with a new keyword function that accepts a strongly typed control, allowing it to be associated with a given binding. This enables users to map compliance controls to Pepr’s bindings, facilitating the generation of compliance reports based on control status.
+
+## Goals
+
+- Pepr to provide the mechanism to run the validation reports.
+- The report generation should be either on-demand or on schedule.
+
+## Non-Goals
+
+- To create a new way to enforce validation outside of the Pepr Modules.
+
+### Consequences ###
+
+- Extend the Capability class with a new function, such as `CompliesWith()`, that accepts a strongly typed control to associate it with a given binding.
+- Store compliance data in `PeprStore` in a new instance to prevent corruption of compliance data through Store API.
```

Comment:
It may be worth annotating more granularly _what_ is intended to be stored. Mappings of `control` <> `status` for each definition or an overall collective status. As well as potentially any evidence.

Reply:
Great point